FerramAerospaceResearch.Base/UnityJobs/OcclusionJobs.cs

```diff
@@ -0,0 +1,381 @@
+            {
+                results[index] = quaternion.identity;
+            }
+            else if (index == points - 1)
+            {
+                results[index] = quaternion.Euler(180, 0, 0);
+            }
+            else
+            {
+                float theta = index * thetaConstantTerm;
+                float denom = points - 1 + (2 * epsilon);
+                float num = 2 * (index + epsilon);
+                float phi = math.acos(1 - num / denom);
+                float3 res = new float3(math.cos(theta) * math.sin(phi),
+                                        math.sin(theta) * math.sin(phi),
+                                        math.cos(phi));
+                results[index] = Quaternion.FromToRotation(new float3(0,0,1), res);
+            }
+        }
+    }
+
+    [BurstCompile]
+    public struct SetBoundsJob : IJobParallelFor
+    {
+        [ReadOnly] public float3 boundsCenter;
+        [ReadOnly] public float3 extents;
+        [ReadOnly] public NativeArray<quaternion> quaternions;
+        [WriteOnly] public NativeArray<float2> bounds;
+
+        // We now have center, extents, and the raycaster plane (normal = rotation * vessel.forward)
+        // Project each of the 8 corners of the bounding box onto the plane.
+        // https://stackoverflow.com/questions/9605556/how-to-project-a-point-onto-a-plane-in-3d/41897378#41897378
+        public void Execute(int index)
+        {
+            float3 normal = math.mul(quaternions[index], new float3(0, 0, 1));
+            float3 origin = float3.zero + math.length(extents) * normal;
+            float2 min = new float2(float.PositiveInfinity, float.PositiveInfinity);
+            float2 max = new float2(float.NegativeInfinity, float.NegativeInfinity);
+            float3 ext = extents / 2;
+            float3 center = boundsCenter;
+            unsafe
+            {
+                float3* arr = stackalloc[]
+                {
+                    center - ext,
+                    new float3(center.x - ext.x, center.y - ext.y, center.z + ext.z),
+                    new float3(center.x - ext.x, center.y + ext.y, center.z - ext.z),
+                    new float3(center.x - ext.x, center.y + ext.y, center.z + ext.z),
+                    new float3(center.x + ext.x, center.y - ext.y, center.z - ext.z),
+                    new float3(center.x + ext.x, center.y - ext.y, center.z + ext.z),
+                    new float3(center.x + ext.x, center.y + ext.y, center.z - ext.z),
+                    center + ext,
+                };
+
+                for (int i = 0; i < 8; i++)
+                {
+                    // Project the bounding box onto the plane
+                    float dist = math.dot(arr[i] - origin, normal);
+                    float3 projection = arr[i] - dist * normal;
+                    // Rotate the projection back towards the vessel alignment
+                    projection = math.mul(math.inverse(quaternions[index]), projection);
```

Comment:
Quaternion inverse can be computed outside the loop

Reply:
Implemented in 53809f6